test/Microsoft.NET.Publish.Tests/GivenThatWeWantToPreserveCompilationContext.cs

```diff
@@ -35,6 +35,7 @@ public void It_publishes_the_project_with_a_refs_folder_and_correct_deps_file()
                     propertyGroup.Should().NotBeNull();
 
                     propertyGroup.Add(new XElement(ns + "DisableImplicitFrameworkReferences", "true"));
+                    propertyGroup.Add(new XElement(ns + "DisableLockFileFrameworks", "true"));
```

Comment:
Can you update the comment to point to https://github.com/Microsoft/msbuild/issues/1345 (as #367 has been closed in favor of that).  Also, these workarounds should only be necessary when targeting .NET Framework, so it would be nice to condition them based on that.